### PR TITLE
Uperf plugin

### PR DESCRIPTION
**Pull request by jaredoconnell**
## Changes introduced with this PR

This PR adds UPerf Support.

The best way to see it operational is with the docker-compose file, as this plugin requires the orchestration of two components.

Sub-tasks:

- [x] Basic Operation
- [x] Multi-staged Dockerfile
- [x] Throughout output
- [x] Test Cases
- [x] Validated Profile Input
- [x] Input Documentation
- [x] README
- [x] Latency Output

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: python/uperf/Dockerfile
Comment (dustinblack) [inline python/uperf/Dockerfile:26]: Needs updated

File: python/uperf/uperf_plugin.py
Comment (dustinblack) [inline python/uperf/uperf_plugin.py:37]: Input schema params need metadata documentation (name and description)

File: python/uperf/uperf_plugin.py
Comment (dustinblack) [inline python/uperf/uperf_plugin.py:74]: It looks like you've already separated the server into its own step.

File: python/uperf/uperf_plugin.py
Comment (dustinblack) [inline python/uperf/uperf_plugin.py:127]: Complete nit pick and functionally inconsequential, but I would put the server step above the client step because it aligns better with the workflow.

File: python/uperf/uperf_plugin.py
Comment (dustinblack) [reply python/uperf/uperf_plugin.py:37]: It looks like we also have some uperf parameter support to add still for MVP. And I wonder if we should consider adding an IP address data type to the SDK.

**Review by dustinblack — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (dustinblack) [reply .gitignore:4]: https://github.com/arcalot/arcaflow-plugins/issues/10

File: python/uperf/Dockerfile
Comment (mfleader) [inline python/uperf/Dockerfile:17]: Remove commented code

**Comment by mfleader** [comment]
I think you should add a readme to your plugin root directory.

File: python/uperf/test_uperf_plugin.py
Comment (mfleader) [inline python/uperf/test_uperf_plugin.py:48]: grammatical error?

Comment (jaredoconnell) [reply python/uperf/test_uperf_plugin.py:48]: I don't think so. The first one is active tense, and the second one is passive tense.

**Comment by mkarg75** [comment]
LGTM!

File: python/uperf/LICENSE
Comment (Harshith-umesh) [inline python/uperf/LICENSE:1]: please remove this license, the plan is to keep one license for all the plugins

File: python/uperf/requirements.txt
Comment (Harshith-umesh) [inline python/uperf/requirements.txt:1]: this needs to be changed before merge

Comment (jaredoconnell) [reply python/uperf/requirements.txt:1]: Correct. I need to wait for my arcaflow-plugin-sdk-python changes to be merged in for that. It's not compatible with the releases.

File: python/uperf/test_uperf_plugin.py
Comment (jaredoconnell) [reply python/uperf/test_uperf_plugin.py:114]: Clarified comment.

File: python/uperf/uperf_plugin.py
Comment (jaredoconnell) [reply python/uperf/uperf_plugin.py]: Documentation done. The address should stay a string, since it could be an ip address, domain, or hostname. For example, in the container, it uses "server" for the addr, which is defined in the docker-compose.yml as the cluster network.

File: python/uperf/uperf_plugin.py
Comment (jaredoconnell) [reply python/uperf/uperf_plugin.py:131]: Looking at the documentation, I don't see any documented exceptions that would be thrown given that I give it no timeout and check=False. What's likely to end up happening is there will be errors in the stdout, which will be detected by the plugin and addressed appropriately.

**Review by dustinblack — APPROVED** [review]
Ship it!

**Comment by jaredoconnell** [comment]
I found one major thing I can improve with the output. It's nearly complete, but needs more testing, so I will have this done tomorrow.

**Comment by dustinblack** [comment]
> I found one major thing I can improve with the output. It's nearly complete, but needs more testing, so I will have this done tomorrow.

@jaredoconnell If the additional work can be converted into an issue (i.e., doesn't fundamentally affect the MVP), let's get this merged now and keep iterating quality and features via future PRs.

**Comment by dustinblack** [comment]
Please squash and rebase, and I think we can get this merged. Thanks!

**Review by dustinblack — APPROVED** [review]

**Pull request merged**
